pulp_ansible/app/tasks/collections.py

```diff
@@ -460,8 +473,59 @@ def _collection_list_downloader(self, api_version, collection_endpoint, page_num
             collection_list_url = f"{collection_endpoint}?limit={page_size}&offset={offset}"
         return self.remote.get_downloader(url=collection_list_url)
 
+    async def _download_unpaginated_repo_metadata(self):
+        self._unpaginated_collection_metadata = None
+        self._unpaginated_collection_version_metadata = None
+
+        root_endpoint, api_version = await self._get_root_api(self.remote.url)
+        self._api_version = api_version
+        collection_endpoint = f"{root_endpoint}/metadata/collections/?deprecated=true"
```

Comment:
Is there any reason the /metadata/ is needed?  This seems like info that would be fitting for Collection and CollectionVersion endpoints?

Reply:
the metadata endpoints are unpaginated, the main difference is getting all metadata with just one request instead of doing a lot of requests

Reply:
@alikins yeah the issue is that the existing endpoints are paginated and also for the CollectionVersion detail view it doesn't contain all of the info needed when the responses come in one large response. Also these endpoints are already merged; this code actually just uses them for syncing. So if we're changing the design that's ok but that wouldn't be in this PR.